File: compatibility-tests/basic/gradle-java/src/test/java/TypeConverterSourcesTest.java
import org.tabletest.junit.TableTest;
import org.tabletest.junit.TypeConverterSources;
import static org.junit.jupiter.api.Assertions.assertEquals;

@TypeConverterSources({YesNoConverter.class})
public class TypeConverterSourcesTest {
    @TableTest("""
        Scenario         | input | expected?
        yes maps to true | yes   | yes
        no maps to false | no    | no
        """)
    void yes_no_converter(boolean input, boolean expected) {
        assertEquals(expected, input);
    }
}

File: compatibility-tests/basic/maven-kotlin/src/test/kotlin/TypeConverterSourcesTest.kt
import org.tabletest.junit.TableTest
import org.tabletest.junit.TypeConverterSources
import kotlin.test.assertEquals

@TypeConverterSources(YesNoConverter::class)
class TypeConverterSourcesTest {
    @TableTest("""
        Scenario         | input | expected?
        yes maps to true | yes   | yes
        no maps to false | no    | no
        """)
    fun yes_no_converter(input: Boolean, expected: Boolean) {
        assertEquals(expected, input)
    }
}
